lucille-core/src/main/java/com/kmwllc/lucille/util/JDBCUtils.java

```diff
@@ -0,0 +1,131 @@
+package com.kmwllc.lucille.util;
+
+import com.kmwllc.lucille.core.Document;
+import java.sql.Date;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class JDBCUtils {
+
+  private static final Logger log = LogManager.getLogger(JDBCUtils.class);
+
+  public static void parseResultToDoc(Document doc, ResultSet rs, Map<String, Object> fieldMapping) throws SQLException {
+    ResultSetMetaData metadata = rs.getMetaData();
+    for (String key : fieldMapping.keySet()) {
+      String field = (String) fieldMapping.get(key);
+      int columnIndex = rs.findColumn(key);
+      int columnType = metadata.getColumnType(columnIndex);
+      typeHandling(doc, rs, field, columnIndex, columnType);
+    }
+  }
+
+  public static void parseResultToDoc(Document doc, ResultSet rs, String fieldName)
+      throws SQLException {
+    int columnIndex = rs.findColumn(fieldName);
+    parseResultToDoc(doc, rs, fieldName, columnIndex);
+  }
+
+  public static void parseResultToDoc(Document doc, ResultSet rs, String fieldName, int columnIndex)
+      throws SQLException {
+    ResultSetMetaData metadata = rs.getMetaData();
+    typeHandling(doc, rs, fieldName, columnIndex, metadata.getColumnType(columnIndex));
+  }
+
+  private static void typeHandling(Document doc, ResultSet rs, String fieldName, int columnIndex, int columnType)
+      throws SQLException {
+      switch (columnType) {
```

Comment:
Great use of the switch statement and the type mappings from SQL -> Java all look good to me. Nice work!